SU2_CFD/include/fluid/CCoolProp.hpp

```diff
@@ -30,27 +30,41 @@
 #if defined(HAVE_COOLPROP) && !defined(CODI_FORWARD_TYPE) && !defined(CODI_REVERSE_TYPE)
 #define USE_COOLPROP
 namespace CoolProp {
-  class AbstractState;
+class AbstractState;
 }
 #endif
 #include <memory>
 
-
 /*!
  * \class CCoolProp
  * \brief Child class for defining fluid model from CoolProp library.
  * \author: P.Yan
  */
 class CCoolProp final : public CFluidModel {
  private:
-  su2double Gamma{1.4};           /*!< \brief Ratio of Specific Heats. */
-  su2double Gas_Constant{297};    /*!< \brief specific Gas Constant. */
-  su2double Pressure_Critical{0.0};   /*!< \brief critical pressure */
-  su2double Temperature_Critical{0.0};    /*!< \brief critical temperature */
+  su2double Gamma{1.4};                /*!< \brief Ratio of Specific Heats. */
+  su2double Gas_Constant{297};         /*!< \brief specific Gas Constant. */
+  su2double Pressure_Critical{0.0};    /*!< \brief critical pressure */
+  su2double Temperature_Critical{0.0}; /*!< \brief critical temperature */
   su2double acentric_factor{0.0};      /*!< \brief acentric factor */
+  su2double dp{0.01};             /*!< threshold for pressure */
 #ifdef USE_COOLPROP
-  std::unique_ptr<CoolProp::AbstractState> fluid_entity;   /*!< \brief fluid entity */
+  std::unique_ptr<CoolProp::AbstractState> fluid_entity; /*!< \brief fluid entity */
 #endif
+  /*!
+   * \brief Avoid critical pressure
+   * \return .
+   */
+  void Check_Pressure(su2double Pressure) {
+    if (abs(Pressure / Pressure_Critical - 1) < dp) {
+      if (Pressure >= Pressure_Critical) {
+        Pressure = Pressure_Critical * (1 + dp);
+      }
+      else {
+        Pressure = Pressure_Critical * (1 - dp);
+      }
+    }
+  }
```

Comment:
This function was not modifying pressure.
```suggestion
  void CheckPressure(su2double& Pressure) const {
    if (Pressure > Pressure_Critical) Pressure = fmax(Pressure, (1 + dp) * Pressure_Critical);
    else Pressure = fmin(Pressure, (1 - dp) * Pressure_Critical);
  }
```

Reply:
this function is modified following your advice.